vm/vtpm-mitigation/src/lib.rs

```diff
@@ -0,0 +1,270 @@
+        *self.pending_nvram.lock() = state.to_vec();
+        tracing::info!("commit_nv_state: {} bytes", state.len());
+        Ok(())
+    }
+
+    fn get_crypt_random(&mut self, buf: &mut [u8]) -> DynResult<usize> {
+        match getrandom::fill(buf) {
+            Ok(()) => Ok(buf.len()),
+            Err(_) => Err(Box::new(PlatformError::ErrorRngGenerator)),
+        }
+    }
+
+    fn monotonic_timer(&mut self) -> std::time::Duration {
+        self.time.elapsed()
+    }
+
+    fn get_unique_value(&self) -> &'static [u8] {
+        b"hvlite vtpm"
+    }
+}
+
+#[allow(non_snake_case)]
+#[expect(unsafe_code)]
+#[unsafe(no_mangle)]
+/// FFI to check if the given blob is a valid TPM blob
+pub extern "C" fn IsValidVtpmBlob(blob: *const u8, size: usize, err_offset: *mut u64) -> u64 {
+    if blob.is_null() || err_offset.is_null() || size == 0 {
+        tracing::error!("Input blob is null");
+        return STATUS_INVALID_POINTER;
+    }
+
+    // SAFETY: The caller should ensure that the input pointer is valid and size is within the bounds
+    let blob = unsafe { std::slice::from_raw_parts(blob, size) };
+    let result = is_valid_tpm_blob(blob);
+    unsafe {
+        *err_offset = 0;
+    }
+    match result {
+        Ok(_) => STATUS_VALID_TPM_STATE,
+        Err(size) => {
+            unsafe {
+                *err_offset = size as u64;
+            }
+            STATUS_INVALID_TPM_STATE
+        },
+    }
+}
+
+#[allow(non_snake_case)]
+#[expect(unsafe_code)]
+#[unsafe(no_mangle)]
+/// FFI to Recover the given blob and return the recovered blob
+pub extern "C" fn RecoverVtpmBlob(blob: *mut u8, size: usize, out_blob: *mut u8, out_blob_size: usize) -> u64 {
+    if blob.is_null() || out_blob.is_null() || size == 0 || out_blob_size == 0 {
+        tracing::error!("Input or output blob is null");
+        return STATUS_INVALID_POINTER;
+    }
+
+    if size != out_blob_size {
+        tracing::error!("Input and output blob size should be same");
+        return STATUS_RECOVERY_INPUT_OUTPUT_BLOB_SIZE_MISMATCH;
+    }
+
+    // SAFETY: The caller should ensure that the input and output blob pointers are valid and size is within the bounds
+    let blob = unsafe { std::slice::from_raw_parts(blob, size) };
+    let out_blob = unsafe { std::slice::from_raw_parts_mut(out_blob, out_blob_size) };
+
+    match recover_vtpm_blob(blob) {
+        Ok(data) => {
+            out_blob.copy_from_slice(&data);
+            STATUS_RECOVERY_SUCCESS
+        }
+        Err(_) => {
+            STATUS_RECOVERY_FAILED
+        }
+    }
+}
+
+/// Check if the given blob is a valid TPM blob
+fn is_valid_tpm_blob(blob: &[u8]) -> Result<(), usize> {
+    let result = check_blob(blob);
+    if let Err(size) = result {
+        Err(size)
+    } else {
+        Ok(())
+    }
+}
+
+/// Recover the given blob and return the recovered blob
+fn recover_vtpm_blob(original_blob: &[u8]) -> Result<Vec<u8>, bool> {
```

Comment:
don't use bool as an error type, make an explicit error type that is clear about what kind of error is being represented here.